pkg/skaffold/runner/dev.go

```diff
@@ -33,14 +33,15 @@ import (
 var ErrorConfigurationChanged = errors.New("configuration changed")
 
 func (r *SkaffoldRunner) doDev(ctx context.Context, out io.Writer) error {
-	r.logger.Mute()
+	actionPerformed := false
 
 	// acquire the intents
 	buildIntent, syncIntent, deployIntent := r.intents.GetIntents()
 
 	if (r.changeSet.needsRedeploy && deployIntent) ||
 		(len(r.changeSet.needsRebuild) > 0 && buildIntent) ||
 		(len(r.changeSet.needsResync) > 0 && syncIntent) {
+		r.logger.Mute()
```

Comment:
why can't we just put a `defer r.logger.Unmute()` here? It seems equivalent.

Reply:
because we don't want to unmute the logs if the build fails. that wouldn't fix the issue

Reply:
ah right, the return statements in the middle threw me off.